src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -138,8 +157,14 @@
     UserRoleAssignmentSchema,
     UserSchema,
 )
+from zenml.zen_stores.schemas.base_schemas import BaseSchema
 
 AnyNamedSchema = TypeVar("AnyNamedSchema", bound=NamedSchema)
+AnySchema = TypeVar("AnySchema", bound=BaseSchema)
+B = TypeVar("B", bound=BaseResponseModel)
```

Comment:
why can't you use `BaseResponseModel` directly in the type annotations? That would make it more readable IMO

Reply:
The logic would still work with `BaseResponseModel`, but the type hints would be much worse:
For example

```python
class Base:
  def do_something():
    pass

class A(Base):
  pass

class B(Base):
  pass

def f1(item: Base) -> Base:
  item.do_something()
  return item

B = TypeVar("B", bound=Base)

def f2(item: B) -> B:
  item.do_something()
  return item

a = A()
f1(a)  # the return value will have type `Base`
f2(a)  # the return value will have type `A`
```